src/composables/transactionList.ts

```diff
@@ -0,0 +1,324 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import {
+  computed, onMounted, ref, watch,
+} from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+import {
+  ITransaction,
+  ITransactionsState, ITransactionStateList,
+} from '../types';
+import { useMiddleware } from './middleware';
+import {
+  AEX9_TRANSFER_EVENT, fetchJson, getLocalStorageItem, setLocalStorageItem, TX_DIRECTION,
+} from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+const LOCAL_STORAGE_KEY = ['transactionsState'];
+
+const isInitalized = ref(false);
+
+const generateEmptyTransactionState = (): ITransactionsState => ({
+  loaded: [],
+  nextPageUrl: '',
+  pending: {},
+  tipWithdrawnTransactions: [],
+});
+
+const transactions = ref<ITransactionStateList>(
+  getLocalStorageItem<ITransactionStateList>(LOCAL_STORAGE_KEY) || {},
+);
+
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  transactions.value[address] = generateEmptyTransactionState();
+
+  return transactions.value[address];
+}
+
+function getTransactionByHash(address: Encoded.AccountAddress, hash: string) {
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  }
+}
+
+const setTransactionsNextPage = (address: Encoded.AccountAddress, url: string) => {
+  transactions.value[address].nextPageUrl = url;
+};
+
+export const useTransactionList = () => {
```

Comment:
Please convert this to regular function as this is the way ow all other composables are written.